### DISCUSSION TITLE
Allow for "id" to be optionally passed into multicall for easier reference

### DISCUSSION THREAD

**Discussion by lootboi** [wevm/viem#2001] [Idea / Feature Request]
## Problem
Working with hundreds of requests, to different contracts, for different reasons, all at once through the `mutlticall` method leads to instances where developers have to use workaround methods to filter out the different results for parsing.

## Solution
Allowing a unique identifier to be passed along with each contract object gives developers an easier way to filter results without significant workaround.

### Current implementation

```typescript
const results = await publicClient.multicall({
contracts: [{ 
     address: <address>,
     abi: <abi>,
     functionName: <functionName>
     }]
})

// allowFailure == true
results = [
  {
  result: <result>,
  status: <status>
  }
]

// allowFailure == false
results = [<result>]
```

### Proposed implementation

```typescript
const results = await publicClient.multicall({
contracts: [{ 
     address: <address>,
     abi: <abi>,
     functionName: <functionName>,
     id: <id string>
     }]
})

// allowFailure == true
results = [
  {
  result: <result>,
  status: <status>,
   id: <id string>
  }
]

// allowFailure == false
results = [{result: <result>, id: <id string>}]
```

**Comment by MidnightLightning** [comment]
This seems related to this discussion: https://github.com/wevm/viem/discussions/1699
